Sources/SmithyReadWrite/WireResponseDocumentBinding.swift

```diff
@@ -0,0 +1,19 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import struct Foundation.Data
+
+public typealias WireResponseDocumentBinding<WireResponse: WireDataProviding, Reader: SmithyReader> =
+    (WireResponse) async throws -> Reader
```

Comment:
This typealias replaces the former `HTTPResponseDocumentBinding` with a closure that is not dependent on HTTP.

Otherwise, this is just moved in from ClientRuntime.